### PR TITLE
Update the TBScreenCapture

### PR DESCRIPTION
**Pull request by Lucashuang0802**
We have been using this component in vertical solution for a couple iOS apps. We have a feature called screen sharing + annotation. We found it easier to achieve cross-platform annotation if we stream the intrinsic size of a UIView. Here is the list of modification:
- remove the resize method that is for fixing an edge limit issue introduced from Chromium: https://bugs.chromium.org/p/webrtc/issues/detail?id=4643#c7
- remove config generated from CocoaPods. I am not able to get it up and running by not removing those setting.
- I have tested it and it seemed okay. But I am not sure due to my webRTC and custom capture knowledge.

### CODE REVIEW

**Comment by Lucashuang0802** [comment]
Adding Cesar as he was the one to discuss this with me and I don't know who should review this: @Nitrillo

File: 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m [hunk 1]
```diff
@@ -110,7 +110,8 @@ - (void)initCapture {
     dispatch_source_set_event_handler(_timer, ^{
         @autoreleasepool {
             __block UIImage* screen = [_self screenshot];
-            CGImageRef paddedScreen = [self resizeAndPadImage:screen];
+//            CGImageRef paddedScreen = [self resizeAndPadImage:screen];
+            CGImageRef paddedScreen = [screen CGImage];
```
Comment (Lucashuang0802) [inline 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m]: This is the critical line of the PR.

File: 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.h [hunk 2]
```diff
@@ -10,12 +10,12 @@
...
 @protocol OTVideoCapture;
 
-// defines for image scaling
-// From https://bugs.chromium.org/p/webrtc/issues/detail?id=4643#c7 :
-// Don't send any image larger than 1280px on either edge. Additionally, don't
-// send any image with dimensions %16 != 0
-#define MAX_EDGE_SIZE_LIMIT 1280.0f
-#define EDGE_DIMENSION_COMMON_FACTOR 16.0f
+//// defines for image scaling
+//// From https://bugs.chromium.org/p/webrtc/issues/detail?id=4643#c7 :
```
Comment (wobbals) [inline 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.h]: double comments?

**Comment by wobbals** [comment]
I personally prefer not to leave huge blocks of commented code; scm is good for keeping the old stuff around if needed.

As far as the function replaced, those changes were put in to deal with a number of corner cases with respect to keeping proper padding for the source image. A customer sent in an example that basically just used a webview in the root view controller and it was obvious that we needed to do some filtering in order to preserve all the pixels being tested. 

Have you had a chance to run these changes through a variety of different frame sizes?

**Comment by Lucashuang0802** [comment]
@wobbals I agree with the huge block. I am going to test on that.

File: 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.h (hunk 2 above)
Comment (Lucashuang0802) [inline 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.h]: Lazy commenter. lol

**Comment by Lucashuang0802** [comment]
@wobbals I was testing on iPhone 6s, 5s, 4s, iPad pro, iPad air 2, iPad retina simulator. The previous distorted does go away. My Chrome browser version is: `Version 56.0.2924.76 (64-bit)`. The idea to remove this piece of code is to let another peer connection to know the exact size of the streaming screen. Previously, the video dimension was modified so it's off.

**Comment by Lucashuang0802** [comment]
Please DO NOT MERGE until someone(probably) me to verify the one in Swift sample 😃

**Comment by Lucashuang0802** [comment]
@chetanvangadiTokbox I was testing on iPhone 6s, 5s, 4s, iPad pro, iPad air 2, iPad retina simulator. The previous distorted does go away. My Chrome browser version is: Version 56.0.2924.76 (64-bit). The idea to remove this piece of code is to let another peer connection to know the exact size of the streaming screen. Previously, the video dimension was modified so it's off.

**Comment by Lucashuang0802** [comment]
the expected testing result will be: there is no distorted screen sharing on another peer connection.

**Comment by chetanvangadiTokbox** [comment]
I tested on iPhone 6 plus, iphone 7 plus and iPad pro, I dont see any distorted image.

File: 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m (hunk 1 above)
Comment (SteveMcFarlin) [inline 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m]: I would remove the commented line of code.

File: 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m [hunk 3]
```diff
@@ -186,122 +187,122 @@ - (int32_t)captureSettings:(OTVideoFormat*)videoFormat
...
+//    CGContextScaleCTM(context, 1.0, -1.0);
+//    CGContextTranslateCTM(context, 0, -destRectForSourceImage.size.height);
+//    CGContextDrawImage(context, destRectForSourceImage, sourceCGImage);
+//    
+//    // Clean up and get the new image.
+//    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+//    UIGraphicsEndImageContext();
+//    
+//    return [newImage CGImage];
+//}
```
Comment (SteveMcFarlin) [inline 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m]: Why is all the code above commented out? If it is no longer used remove it.

**Comment by SteveMcFarlin** [comment]
@Lucashuang0802 If there is code that is no longer used please remove it rather than commenting it out. The critical line change LGTM, but I think someone on the native team or @wobbals should probably sign off on it. It has been a while sense I worked on iOS.

File: 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m [hunk 4]
```diff
@@ -110,7 +110,7 @@ - (void)initCapture {
     dispatch_source_set_event_handler(_timer, ^{
         @autoreleasepool {
             __block UIImage* screen = [_self screenshot];
-            CGImageRef paddedScreen = [self resizeAndPadImage:screen];
+            CGImageRef paddedScreen = [screen CGImage];
```
Comment (Lucashuang0802) [inline 4.Screen-Sharing/Screen-Sharing/TBScreenCapture.m]: This is the critical line of the change.

**Comment by chetanvangadiTokbox** [comment]
@Lucashuang0802 We see distorted image with non standard resolution like 1282 x 720 with this merge, do not merge it

**Pull request closed without merging**